torch/lib/THD/master_worker/common/CommandChannel.cpp

```diff
@@ -84,19 +87,58 @@ bool MasterCommandChannel::init() {
 }
 
 void MasterCommandChannel::sendMessage(std::unique_ptr<rpc::RPCMessage> msg, int rank) {
+  // Throw error which we get from worker.
+  if (thd::master::THDState::s_error != "") {
```

Comment:
I'd make it a `unique_ptr`. Null checks looks cleaner than empty string comparison